secure_timesheet/models/account_analytic_line.py

```diff
@@ -2,25 +2,61 @@
 # © 2017 Savoir-faire Linux
 # License LGPL-3.0 or later (http://www.gnu.org/licenses/LGPL).
 
-from openerp import api, models
+from openerp import api, fields, models, _
+from openerp.exceptions import ValidationError
 
 
 class AccountAnalyticLine(models.Model):
 
     _inherit = 'account.analytic.line'
 
+    # TODO: should we protect project.project? protect analytic lines
+    # attached to projects in a certain state?
+    sheet_id = fields.Many2one(ondelete="restrict")
+
+    @api.multi
+    def check_unlink_access(self):
+        for line in self:
+            sheet = line.sheet_id
+            # TODO: what do we do if there is no associated timesheet?
```

Comment:
simple answer, we do nothing. If the line is not bound to a timesheet, then it has no workflow, so no possible way to know if it should be deleted.